ggshield/core/client.py

```diff
@@ -118,3 +122,38 @@ def check_client_api_key(client: GGClient) -> None:
         raise UnexpectedError(
             f"GitGuardian server is not responding as expected.\nDetails: {response.detail}"
         )
+
+    # Check token scopes if required_scopes is not empty
+    if required_scopes:
+        response = client.api_tokens()
+
+        if not isinstance(response, (Detail, APITokensResponse)):
+            raise UnexpectedError("Unexpected api_tokens response")
+        elif isinstance(response, Detail):
+            raise UnexpectedError(response.detail)
+
+        missing_scopes = required_scopes - set(
+            TokenScope(scope) for scope in response.scopes
+        )
+        if missing_scopes:
+            raise MissingScopesError(list(missing_scopes))
+
+
+def get_required_token_scopes_from_config(
```

Comment:
It's a good idea to factorize this code, but I would prefer if it were kept in the secret vertical. Maybe you can move this function in verticals/secret_scanner.py?